src/mover_manager.erl

```diff
@@ -325,17 +325,14 @@ handle_sync_event(status, _From, StateName, #state{live_worker_count = LW,
                {objects_in_progress, LW},
                {objects_failed, EC},
                {fatal_stop, FS}],
-    case StateName of
-        % Fixed a race condition that prevents the gen_fsm from transitioning
-        % from the halting state to the ready state when looping over status.
-        %
-        % The zero timeout prevents a status call from clearing the gen_fsm timeout,
-        % and the halting -> ready transition is dependant on a timeout to occur.
-        halting ->
-            {reply, {ok, Summary}, StateName, State, 0};
-        _ ->
-            {reply, {ok, Summary}, StateName, State}
-    end;
+
+    %% Fixed a race condition that prevents the gen_fsm from transitioning
+    %% from the current state to the next state when looping over status.
+    %%
+    %% The zero timeout prevents a status call from clearing the gen_fsm timeout,
+    %% and the state transition is dependant on a timeout to occur.
+    {reply, {ok, Summary}, StateName, State, 0};
```

Comment:
We shouldn't send a 0 timeout to the `ready` state, since it has no timeout handler.